Esbuild and fastify-swagger-ui build output can't access the swagger-ui-dist static files.

### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported


### Fastify version

4.17.0

### Plugin version

1.8.1

### Node.js version

18.16.0

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

12.6.5 

### Description

Attempting to run code built with esbuild, fastify, fastify-swagger-ui results in the error:
```
node:internal/modules/cjs/loader:1075
  const err = new Error(message);
              ^

Error: Cannot find module './static/logo.svg'
Require stack:
- /Users/david/workspace/fastifySwaggerTest/dist/index.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
    at Function.resolve (node:internal/modules/cjs/helpers:116:19)
    at NE (file:///Users/david/workspace/fastifySwaggerTest/dist/index.js:533:3585)
    at lo.exec (file:///Users/david/workspace/fastifySwaggerTest/dist/index.js:17:1196)
    at Et.om (file:///Users/david/workspace/fastifySwaggerTest/dist/index.js:17:2907)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '/Users/david/workspace/fastifySwaggerTest/dist/index.js' ]
}
```

Supplying a valid logo will run the server, but the assets served at static will 404.  

In the provided esbuild.config.mjs, we are defining __filename and __dirname for integration with commonjs libraries.  

### Steps to Reproduce

[Clone repo](https://github.com/drcampbell/fastifyEsbuildError)

run `yarn start` and observe that it doesn't error
run `yarn buildandrun` and observe that it errors


### Expected Behavior

_No response_

--- a/README.md
+++ b/README.md
@@ -321,8 +321,7 @@ Note that this behavior is disabled by default in `@fastify/swagger-ui`.
 
 ### Bundling
 
-If you want to bundle Swagger UI with your application, you need to copy the swagger-ui static files to your server yourself and set the `baseDir` option to point to your folder.
-
+To bundle Swagger UI with your application, the swagger-ui static files need to be copied to the server and the `baseDir` option set to point to the file directory.
 <details>
 <summary>Copy files with esbuild</summary>
 

--- a/README.md
+++ b/README.md
@@ -319,6 +319,51 @@ fastify.register('@fastify/swagger-ui', {
 
 Note that this behavior is disabled by default in `@fastify/swagger-ui`.
 
+### Bundling
+
+If you want to bundle Swagger UI with your application, you need to copy the swagger-ui static files to your server yourself and set the `baseDir` option to point to your folder.
+
+<details>
+<summary>Copy files with esbuild</summary>
+
+```js
+import { build } from 'esbuild'
+import { copy } from 'esbuild-plugin-copy'
+
+await build({
+  // ...
+  plugins: [
+    copy({
+      resolveFrom: 'cwd',
+      assets: {
+        from: ['node_modules/@fastify/swagger-ui/static/*'],
+        to: ['dist/static'],
+      },
+    }),
+  ],
+})
+```
+
+</details>
+
+<details>
+<summary>Copy files with docker</summary>
+
+```Dockerfile
+COPY ./node_modules/@fastify/swagger-ui/static /app/static
+```
+
+</details>
+
+#### Configure Swagger UI to use a custom baseDir
+Set the `baseDir` option to point to your folder.
+
+```js
+await fastify.register(require('@fastify/swagger-ui'), {
+  baseDir: isDev ? undefined : path.resolve('static'),
+})
+```
+
 <a name="license"></a>
 ## License
 

--- a/lib/routes.js
+++ b/lib/routes.js
@@ -202,7 +202,7 @@ function fastifySwagger (fastify, opts, done) {
 
   // serve swagger-ui with the help of @fastify/static
   fastify.register(fastifyStatic, {
-    root: path.join(__dirname, '..', 'static'),
+    root: opts.baseDir || path.join(__dirname, '..', 'static'),
     prefix: staticPrefix,
     decorateReply: false
   })

--- a/test/route.test.js
+++ b/test/route.test.js
@@ -373,6 +373,33 @@ test('/documentation/static/:file should send back the correct file', async (t) 
   }
 })
 
+test('/documentation/static/:file should send back file from baseDir', async (t) => {
+  t.plan(2)
+  const fastify = Fastify()
+  
+  const uiConfig = {
+    baseDir: resolve(__dirname, '..', 'examples', 'static')
+  }
+
+  await fastify.register(fastifySwagger, swaggerOption)
+  await fastify.register(fastifySwaggerUi, uiConfig)
+
+  {
+    const res = await fastify.inject({
+      method: 'GET',
+      url: '/documentation/static/example-logo.svg'
+    })
+    t.equal(res.statusCode, 200)
+    t.equal(
+      res.payload,
+      readFileSync(
+        resolve(__dirname, '..', 'examples', 'static', 'example-logo.svg'),
+        'utf8'
+      )
+    )
+  }
+})
+
 test('/documentation/static/:file 404', async (t) => {
   t.plan(2)
   const fastify = Fastify()

--- a/test/route.test.js
+++ b/test/route.test.js
@@ -376,7 +376,7 @@ test('/documentation/static/:file should send back the correct file', async (t) 
 test('/documentation/static/:file should send back file from baseDir', async (t) => {
   t.plan(2)
   const fastify = Fastify()
-  
+
   const uiConfig = {
     baseDir: resolve(__dirname, '..', 'examples', 'static')
   }

